src/robot/parsing/model/statements.py

```diff
@@ -971,26 +971,35 @@ class WhileHeader(Statement):
     type = Token.WHILE
 
     @classmethod
-    def from_params(cls, condition, indent=FOUR_SPACES, separator=FOUR_SPACES, eol=EOL):
+    def from_params(cls, condition, limit=None, indent=FOUR_SPACES,
+                    separator=FOUR_SPACES, eol=EOL):
         tokens = [Token(Token.SEPARATOR, indent),
                   Token(cls.type),
                   Token(Token.SEPARATOR, separator),
-                  Token(Token.ARGUMENT, condition),
-                  Token(Token.EOL, eol)]
+                  Token(Token.ARGUMENT, condition)]
+        if limit:
+            tokens.extend([Token(Token.SEPARATOR, indent),
+                           Token(Token.OPTION, limit)])
+        tokens.append(Token(Token.EOL, eol))
         return cls(tokens)
 
     @property
     def condition(self):
-        values = self.get_values(Token.ARGUMENT)
-        if len(values) != 1:
-            return ', '.join(values) if values else None
-        return values[0]
+        return ', '.join(self.get_values(Token.ARGUMENT))
+
+    @property
+    def limit(self):
+        value = self.get_value(Token.OPTION)
+        return value.replace('limit=', '') if value else None
 
     def validate(self, context):
-        conditions = len(self.get_tokens(Token.ARGUMENT))
-        if conditions == 0:
+        values = self.tokens[1:]
```

Comment:
I think tokens contain also separators when when parsing with `data_only=False` and this doesn't works. It seems the old logic to validate the number of conditions would work just fine. Unit test should also be added to make sure parsing works with `data_only=False`.